src/repl.rs

```diff
@@ -269,12 +269,16 @@ impl<F: LurkField> ReplState<F> {
         println!("Read from {}: {}", path.as_ref().to_str().unwrap(), input);
         let mut chars = input.chars().peekmore();
 
-        while self
-            .handle_form(store, &mut chars, package, p, update_env)
-            .is_ok()
-        {}
-
-        Ok(())
+        loop {
+            if let Err(e) = self.handle_form(store, &mut chars, package, p, update_env) {
+                if let Some(ParserError::NoInput) = e.downcast_ref::<ParserError>() {
+                    // It's ok, it just means we've hit the EOF
```

Comment:
What will this do with an incomplete expression, like
```
(let ((a 1))
  (+ a 9) ; note no closing paren
```

Reply:
```
~/workspace/lurk-rs $ cargo run foo.lurk 
    Finished dev [unoptimized + debuginfo] target(s) in 0.69s
     Running `target/debug/lurkrs foo.lurk`
Lurk REPL welcomes you.
Running from foo.lurk.
Read from foo.lurk: (+ (+ 1 1) 1

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Syntax error: premature end of input', /home/user/workspace/lurk-rs/src/repl.rs:85:65
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
~/workspace/lurk-rs $ 
```

Reply:
Cool, which I think is as expected, right? Thanks.

Reply:
Yes, "premature end of input" tells you what is wrong, more or less.

I mean, it shouldn't be a panic, but we're going to address that whole can of worms separately.